lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -6153,7 +6153,16 @@ namespace Js
     {
         Assert(scriptContext->GetConfig()->IsES6GeneratorsEnabled());
         DynamicType* generatorType = CreateGeneratorType(prototype);
-        return RecyclerNew(this->GetRecycler(), JavascriptGenerator, generatorType, args, scriptFunction);
+#if GLOBAL_ENABLE_WRITE_BARRIER
+        if (CONFIG_FLAG(ForceSoftwareWriteBarrier))
+        {
+            return RecyclerNewFinalized(this->GetRecycler(), JavascriptGenerator, generatorType, args, scriptFunction);
+        }
+        else
+#endif
+        {
+            return RecyclerNew(this->GetRecycler(), JavascriptGenerator, generatorType, args, scriptFunction);
+        }
```

Comment:
good idea, let me make the change in next commit. 

---
In reply to: [95430032](https://github.com/Microsoft/ChakraCore/pull/2293#discussion_r95430032) [](ancestors = 95430032)